craft_parts/packages/rpm.py

```diff
@@ -0,0 +1,200 @@
+# Copyright 2023 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Support for RPM files."""
+
+import functools
+import logging
+import os
+import subprocess
+from typing import List, Set
+
+from craft_parts.utils import os_utils
+
+from . import errors
+from .base import BaseRepository
+
+logger = logging.getLogger(__name__)
+
+
+class RPMRepository(BaseRepository):
```

Comment:
I know this is an MVP but can we test this class somehow? How about, for example, an integrated test that mocks the OsRelease and the `yum` calls and then does a full lifecycle with a `build-package`? Would it work?

Reply:
FTR, decided to maybe do this in a future (not for this PR).